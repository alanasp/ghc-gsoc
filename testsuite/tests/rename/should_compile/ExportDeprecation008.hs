{-# OPTIONS_GHC -fwarn-warnings-deprecations #-}

-- Test deprecation of variable and data constructor exports

module ShouldCompile where

import ExportDeprecation008_A
import ExportDeprecation008_B

-- func_C is deprecated at the defining module
-- and module ExportDeprecation008_B but not ExportDeprecation008_A, so we expect
-- only one warning

main = ExportDeprecation008_B.func_C

main2 = func_C
