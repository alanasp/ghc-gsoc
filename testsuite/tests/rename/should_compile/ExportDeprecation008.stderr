ExportDeprecation008.hs:14:8: warning: [-Wdeprecations (in -Wdefault)]
    In the use of ‘func_C’
    (imported from ExportDeprecation008_B, but defined in ExportDeprecation008_C):
    Deprecated: "ExportDeprecation008_C shall be imported directly"
